### PR TITLE
08124 - Tipset - ignore events for nodes not in the address book

### LINKED ISSUE #8124: Tipset Algorithm - Don't use nodes not in the address book.
When a node has been removed there may be events still in the state from the old node.  The tipset algorithm currently uses these events as potential parents.  Retrieving the address from the new address book fails.   First check the address book for whether or not the node exists and don't use it if it is no longer in the address book.

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
Fixes #8124

### CODE REVIEW

**Review by alittley — DISMISSED** [review]

**Review by cody-littley — DISMISSED** [review]
I told you it would be simple. 😄

**Comment by cody-littley** [comment]
I'm wondering if it might be good to add a unit test though. Something simple.

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/tipset/TipsetEventCreatorImplTests.java [hunk 1]
```diff
@@ -787,4 +787,67 @@ void frozenEventCreationBug() {
...
+                .build();
+
+        final FakeTime time = new FakeTime();
+
+        final NodeId nodeA = addressBook.getNodeId(0); // self
+        final NodeId nodeB = addressBook.getNodeId(1);
+        final NodeId nodeC = addressBook.getNodeId(2);
+        final NodeId nodeD = addressBook.getNodeId(3);
+        // Node 4 (E) is not in the address book.
+        final NodeId nodeE = new NodeId(4);
```
Comment (cody-littley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/tipset/TipsetEventCreatorImplTests.java:813]: The random address book generator is not using contiguous node IDs, so node 4 might actually be a valid ID. 
```suggestion
        final NodeId nodeE = new NodeId(nodeD.id() + 1);
```

**Review by cody-littley — DISMISSED** [review]

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetEventCreatorImpl.java [hunk 2]
```diff
@@ -126,6 +129,7 @@ public TipsetEventCreatorImpl(
         this.selfId = Objects.requireNonNull(selfId);
         this.transactionSupplier = Objects.requireNonNull(transactionSupplier);
         this.softwareVersion = Objects.requireNonNull(softwareVersion);
+        this.addressBook = Objects.requireNonNull(addressBook);
 
         Objects.requireNonNull(addressBook);
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetEventCreatorImpl.java:134]: duplicate

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetEventCreatorImpl.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetEventCreatorImpl.java]: removed

**Review by cody-littley — APPROVED** [review]

**Pull request merged**